File: ingest/sportsbook_policy.py
"""User-selected sportsbook universe for MLB, Tennis, CFB and NFL."""
BOOKMAKER_KEYS = ("pinnacle", "fanduel", "fanatics", "draftkings", "williamhill_us", "betmgm")
BOOKMAKERS = ",".join(BOOKMAKER_KEYS)

def requested_bookmakers(requested: str | None = None) -> str:
    if not requested:
        return BOOKMAKERS
    keys = {"williamhill_us" if key.strip() == "caesars" else key.strip() for key in requested.split(",")}
    selected = [key for key in BOOKMAKER_KEYS if key in keys]
    if not selected:
        raise ValueError("No configured sportsbooks in requested bookmaker subset")
    return ",".join(selected)

def selected_books(books: dict | None) -> dict:
    source = dict(books or {})
    if "caesars" in source and "williamhill_us" not in source:
        source["williamhill_us"] = source["caesars"]
    return {key: source[key] for key in BOOKMAKER_KEYS if key in source}


def selected_event(event: dict) -> dict:
    return {**event, "bookmakers": [book for book in event.get("bookmakers", []) if book.get("key") in BOOKMAKER_KEYS]}

File: web/src/lib/sportsbook-policy.ts
// Keep aligned with ingest/sportsbook_policy.py; parity is regression-tested.
export const SPORTSBOOK_KEYS = ["pinnacle", "fanduel", "fanatics", "draftkings", "williamhill_us", "betmgm"];
export const SPORTSBOOK_QUERY = SPORTSBOOK_KEYS.join(",");
export const SPORTSBOOK_NAMES: Record<string,string> = {pinnacle:"Pinnacle", fanduel:"FanDuel", fanatics:"Fanatics", draftkings:"DraftKings", williamhill_us:"Caesars", betmgm:"BetMGM"};
export function selectedSportsbooks<T>(books: Record<string,T> | null | undefined): Record<string,T> {
 const source = {...books};
 if ("caesars" in source && !("williamhill_us" in source)) source.williamhill_us = source.caesars;
 return Object.fromEntries(SPORTSBOOK_KEYS.filter(key=>key in source).map(key=>[key,source[key]]));
}
